photon-api/src/main/scala/com/linkedin/photon/ml/projector/IndexMapProjectorRDD.scala

```diff
@@ -155,16 +156,51 @@ protected[ml] class IndexMapProjectorRDD private (indexMapProjectorRDD: RDD[(Str
 }
 
 object IndexMapProjectorRDD {
+
   /**
    * Generate index map based RDD projectors.
    *
    * @param randomEffectDataSet The input random effect data set
    * @return The generated index map based RDD projectors
    */
-  protected[ml] def buildIndexMapProjector(randomEffectDataSet: RandomEffectDataSet): IndexMapProjectorRDD = {
-    val indexMapProjectors = randomEffectDataSet.activeData.mapValues(localDataSet =>
-      IndexMapProjector.buildIndexMapProjector(localDataSet.dataPoints.map(_._2.features))
-    )
+  protected[ml] def buildIndexMapProjector(
+      randomEffectDataSet: RandomEffectDataSet): IndexMapProjectorRDD = {
+
+    val originalSpaceDimension = randomEffectDataSet
+      .activeData
+      .map { case (_, ds) => ds.dataPoints.head._2.features.length }
+      .take(1)(0)
+
+    // Collect active indices for the active dataset
+    val activeIndices = randomEffectDataSet
+      .activeData
+      .mapValues(ds => ds
+        .dataPoints
+        .map(_._2.features)
+        .flatMap(VectorUtils.getActiveIndices)
+        .toSet)
+
+    // Collect active indices for the passive dataset
```

Comment:
passive?

Reply:
This is jargon we use in GAME. The passive dataset is part that's been held back from training so that we limit the impact a single RE problem has on a node. So the active dataset is used for training, and then active + passive are used for scoring.

Reply:
I only meant that this second loop seems to be over passive data, so it looks like "active" in this second comment is a typo :-)

Reply:
Nope, active in this case means the active indices of the vector. :-)